### PR TITLE
Bdd tests

### PR DESCRIPTION
**Pull request by FanniFarkas**

### CODE REVIEW

File: src/test/java/DrukmakoriSivatag/FixPumpSteps.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package test.java.DrukmakoriSivatag;
+
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import main.java.DrukmakoriSivatag.Pipe;
```
Comment (andrasig-bme) [inline src/test/java/DrukmakoriSivatag/FixPumpSteps.java]: Sonar: unused import

File: src/test/java/DrukmakoriSivatag/FixPumpSteps.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import main.java.DrukmakoriSivatag.Plumber;
+import main.java.DrukmakoriSivatag.Pump;
+import main.java.DrukmakoriSivatag.Saboteur;
```
Comment (andrasig-bme) [inline src/test/java/DrukmakoriSivatag/FixPumpSteps.java]: Sonar: unused import

**Review by andrasig-bme — CHANGES_REQUESTED** [review]
Please remove the marked unused imports.

File: src/test/java/DrukmakoriSivatag/FixPumpSteps.java (hunk 1 above)
Comment (FanniFarkas) [reply src/test/java/DrukmakoriSivatag/FixPumpSteps.java]: Done

**Review by andrasig-bme — APPROVED** [review]
Nice job!

File: src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DamageFixPipeSteps {
+    private Saboteur saboteur;
+    private Pipe pipe;
+
+    private Pump pump;
+    private Plumber plumber;
+    @Given("^Saboteur is  on a damageable pipe$")
+    public void saboteur_is_on_a_damageable_pipe(){
```
Comment (BGergo03) [inline src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java]: A single space should separate the right parenthesis and the left brace

File: src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        saboteur.damagePipe();
+    }
+
+    @When("^Plumber fixes the pipe$")
+    public void plumber_fixes_the_pipe() {
+        plumber.fix();
+    }
+    @Then("^The pipe is damaged$")
+    public void the_pipe_is_damaged()  {
+        boolean isDamaged= pipe.isDamaged();
```
Comment (BGergo03) [inline src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java]: This should look like this: boolean isDamaged = pipe.isDamaged();

File: src/test/java/DrukmakoriSivatag/FixPumpSteps.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @When("^Plumber fixes the pump$")
+    public void plumber_fixes_the_pump() {
+        plumber.fix();
+    }
+    @Then("^The pump is not damaged$")
+    public void the_pump_is_not_damaged()  {
+        boolean isDamaged= pump.getIsDamaged();
+        assertEquals(false, isDamaged);
```
Comment (BGergo03) [inline src/test/java/DrukmakoriSivatag/FixPumpSteps.java]: Use assertFalse and assertTrue when checking a boolean value instead

**Review by naslevente — APPROVED** [review]
No additional observations... looks good

File: src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java (hunk 3 above)
Comment (FanniFarkas) [reply src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java]: Done

File: src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java (hunk 4 above)
Comment (FanniFarkas) [reply src/test/java/DrukmakoriSivatag/DamageFixPipeSteps.java]: Done

File: src/test/java/DrukmakoriSivatag/FixPumpSteps.java (hunk 5 above)
Comment (FanniFarkas) [reply src/test/java/DrukmakoriSivatag/FixPumpSteps.java]: Done

**Pull request merged**